File: build-docs-data.js
const fs = require('fs');
const path = require('path');

const docsDir = path.join(__dirname, 'docs');
const outputFile = path.join(__dirname, 'docs-data.js');

let content = 'window.docsContent = {\n';

const files = fs.readdirSync(docsDir);

for (const file of files) {
    if (file.endsWith('.md')) {
        const filePath = path.join(docsDir, file);
        const fileContent = fs.readFileSync(filePath, 'utf8');
        // Escape backticks and standard escape characters
        const escapedContent = fileContent.replace(/\\/g, '\\\\').replace(/`/g, '\\`').replace(/\$/g, '\\$');
        content += `  "docs/${file}": \`${escapedContent}\`,\n`;
    }
}

content += '};\n';

fs.writeFileSync(outputFile, content);
console.log('docs-data.js generated successfully.');

File: build-docs-data.py
import os

docs_dir = 'docs'
output_file = 'docs-data.js'

content = 'window.docsContent = {\n'

for file_name in os.listdir(docs_dir):
    if file_name.endswith('.md'):
        file_path = os.path.join(docs_dir, file_name)
        with open(file_path, 'r', encoding='utf-8') as f:
            file_content = f.read()
        
        escaped_content = file_content.replace('\\', '\\\\').replace('`', '\\`').replace('$', '\\$')
        content += f'  "docs/{file_name}": `{escaped_content}`,\n'

content += '};\n'

with open(output_file, 'w', encoding='utf-8') as f:
    f.write(content)

print('docs-data.js generated successfully.')
